airbyte-server/src/main/java/io/airbyte/server/ConfigDumpExporter.java

```diff
@@ -145,4 +167,77 @@ private static Path buildConfigPath(final Path storageRoot, final String schemaT
         .resolve(String.format("%s.yaml", schemaType));
   }
 
+  public File exportWorkspace(UUID workspaceId) throws JsonValidationException, IOException, ConfigNotFoundException {
+    final Path tempFolder = Files.createTempDirectory(Path.of("/tmp"), ARCHIVE_FILE_NAME);
+    final File dump = Files.createTempFile(ARCHIVE_FILE_NAME, ".tar.gz").toFile();
+    exportVersionFile(tempFolder);
+    exportConfigsDatabase(tempFolder, workspaceId);
+
+    Archives.createArchive(tempFolder, dump.toPath());
+    return dump;
+  }
+
+  private void exportConfigsDatabase(Path parentFolder, UUID workspaceId) throws IOException, JsonValidationException, ConfigNotFoundException {
```

Comment:
at a high level it feels like there should be an `export(list<uuid> workspaceIds)` instead of having two different implementations one scoped by workspace. Would that make sense here or am I missing something?

Reply:
This export function scoped by workspace Id is exporting configurations related to a workspace and then "cascade" down to other configurations that are being referred/dependent on them.

The second export without scope is dumping everything regardless of links/relations between configurations and thus, will cover more configurations than this `export(uuid)` or a `export(list<uuid>)` function